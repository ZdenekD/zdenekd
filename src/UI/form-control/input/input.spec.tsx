import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../../__test__/utils/helpers';
import Input from '.';

describe('Input', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Input name="input" label="Input" />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-input');

        expect(component.exists()).toBe(true);
    });

    it('renders error', () => {
        const container = shallow(<Input name="input" label="Input" error="Error message" />);
        const component = findComponent(container, 'component-input-error');

        expect(component.exists()).toBe(true);
    });

    it('set length on change', () => {
        const mockSetState = jest.fn();
        const value = 'value';
        const mockEvent = {target: {value}};

        React.useState = jest.fn(() => [0, mockSetState]);

        const container = shallow(<Input name="input" label="Input" />);
        const component = findComponent(container, 'component-input');

        component.simulate('change', mockEvent);

        expect(mockSetState).toHaveBeenCalledWith(value.length);
    });
});
