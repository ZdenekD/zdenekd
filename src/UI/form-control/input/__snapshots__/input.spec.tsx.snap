// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input match snapshot 1`] = `
<div class="control ">
  <label for="_34rFI1"
         class="label "
  >
    Input
  </label>
  <div class="wrapper">
    <input type="text"
           id="_34rFI1"
           name="input"
           class="input  "
           data-test="component-input"
    >
  </div>
</div>
`;
